lxd/network/driver_ovn.go

```diff
@@ -214,6 +216,87 @@ func (n *ovn) projectRestrictedSubnets(p *api.Project, uplinkNetworkName string)
 	return projectRestrictedSubnets, nil
 }
 
+func (n *ovn) randomExternalAddress(ctx context.Context, ipVersion int, uplinkRoutes []*net.IPNet, projectRestrictedSubnets []*net.IPNet, validator func(*net.IPNet) error) (*net.IPNet, error) {
+	// Ensure a sensible deadline is set.
+	_, hasDeadline := ctx.Deadline()
+	var cancel context.CancelFunc = func() {}
+	if !hasDeadline {
+		ctx, cancel = context.WithTimeout(ctx, 5*time.Second)
+	}
+
+	defer cancel()
+
+	var subnets []*net.IPNet
+	for _, projectRestrictedSubnet := range projectRestrictedSubnets {
+		if (projectRestrictedSubnet.IP.To4() != nil && ipVersion == 4) || (projectRestrictedSubnet.IP.To4() == nil && ipVersion == 6) {
+			subnets = append(subnets, projectRestrictedSubnet)
+		}
+	}
+
+	if projectRestrictedSubnets == nil {
+		for _, uplinkRoute := range uplinkRoutes {
+			if (uplinkRoute.IP.To4() != nil && ipVersion == 4) || (uplinkRoute.IP.To4() == nil && ipVersion == 6) {
+				subnets = append(subnets, uplinkRoute)
+			}
+		}
+	}
+
+	if len(subnets) == 0 {
```

Comment:
Feels like these 2 stanzas should be a case statement, it looks a bit odd calling `len` twice so close together.